sklearn/preprocessing/mice.py

```diff
@@ -0,0 +1,359 @@
+
+    verbose : boolean (default = False)
+        Whether to provide updates on progress.
+
+    random_state : int seed, RandomState instance (default = None)
+        The seed of the pseudo random number generator to use when
+        shuffling the data.
+
+    Notes
+    -----
+    Columns which only contain missing values at `fit` are discarded upon
+        `transform`.
+
+    References
+    ----------
+    .. [1] `Stef van Buuren, Karin Groothuis-Oudshoorn (2011). "mice:
+        Multivariate Imputation by Chained Equations in R". Journal of
+        Statistical Software 45: 1-67.
+        <https://www.jstatsoft.org/article/view/v045i03>`_
+    """
+
+    def __init__(
+            self,
+            missing_values='NaN',
+            imputation_order='monotone',
+            n_imputations=100,
+            n_burn_in=10,
+            n_nearest_columns=np.infty,
+            initial_fill_method="mean",
+            min_value=-np.infty,
+            max_value=np.infty,
+            verbose=False,
+            random_state=None):
+        from ..linear_model import BayesianRidge  # avoiding circular import
+        self.model = BayesianRidge()
+        self.missing_values = missing_values
+        self.imputation_order = imputation_order
+        self.n_imputations = n_imputations
+        self.n_burn_in = n_burn_in
+        self.n_nearest_columns = n_nearest_columns
+        self.initial_fill_method = initial_fill_method
+        self.min_value = min_value
+        self.max_value = max_value
+        self.verbose = verbose
+        self.random_state = random_state
+
+    def _fill_in_one_column(self,
+                            X_filled,
+                            mask_missing_values,
+                            this_column,
+                            other_columns,
+                            model=None,
+                            min_std=1e-6):
+        """
+        This function predicts the missing values of one of the
+        features using the current estimates of all the other
+        features. The `model` must support `return_std=True`
+        in its `predict` method for this function to work.
+        """
+
+        # if nothing is missing, just return the default
+        rng = check_random_state(self.random_state)
+        if mask_missing_values[:, this_column].sum() == 0:
+            return X_filled, model
+        missing_row_mask = mask_missing_values[:, this_column]
+        if model is None:
+            X_train = X_filled[:, other_columns][~missing_row_mask]
+            y_train = X_filled[:, this_column][~missing_row_mask]
+            if np.std(y_train) > 0:
+                model = clone(self.model)
+                model.fit(X_train, y_train)
+            else:
+                model = DummyRegressor()
+                model.fit(X_train, y_train)
+        X_test = X_filled[:, other_columns][missing_row_mask]
+        mus, sigmas = model.predict(X_test, return_std=True)
```

Comment:
I am not a big fan of it. I would prefer something which use the `inspect` module with [`signature`](https://docs.python.org/3/library/inspect.html#inspect.signature) --- there is a backport in externals I am pretty sure --- to find if the argument if required by the function.